controllers/deactivation/deactivation_controller_test.go

```diff
@@ -243,47 +456,159 @@ func TestReconcile(t *testing.T) {
+							require.True(t, states.Deactivated(userSignupFoobar))
+
+							t.Run("usersignup already deactivated", func(t *testing.T) {
+								// additional reconciles should find the usersignup is already deactivated
+								res, err := r.Reconcile(context.TODO(), req)
+								// then
+								require.NoError(t, err)
+								require.False(t, res.Requeue, "requeue should not be set")
+								require.True(t, res.RequeueAfter == 0, "requeueAfter should not be set")
+							})
 						})
 					})
 				})
 			})
+
+			t.Run("usersignup should be deactivated - other tier (60 days)", func(t *testing.T) {
+				states.SetDeactivating(userSignupFoobar, true)
+				// given
+				murProvisionedTime := &metav1.Time{Time: time.Now().Add(-time.Duration(expectedDeactivationTimeoutOtherTier*24) * time.Hour)}
+				mur := murtest.NewMasterUserRecord(t, username, murtest.TierName(otherTier.Name), murtest.Account("cluster1"), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
+				mur.Labels[toolchainv1alpha1.MasterUserRecordOwnerLabelKey] = userSignupFoobar.Name
+				r, req, cl := prepareReconcile(t, mur.Name, otherTier, mur, userSignupFoobar, config)
+				// when
+				res, err := r.Reconcile(context.TODO(), req)
+				// then
+				require.NoError(t, err)
+				require.False(t, res.Requeue, "requeue should not be set")
+				require.True(t, res.RequeueAfter == 0, "requeue should not be set")
+				assertThatUserSignupDeactivated(t, cl, username, true)
+				AssertMetricsCounterEquals(t, 1, metrics.UserSignupAutoDeactivatedTotal)
+			})
 		})
+	})
 
-		// the time since the mur was provisioned exceeds the deactivation timeout period for the 'other' tier
-		t.Run("usersignup should be deactivated - other tier (60 days)", func(t *testing.T) {
-			states.SetDeactivating(userSignupFoobar, true)
+	t.Run("test usersignup deactivating state reset to false", func(t *testing.T) {
+		t.Run("when the provisioned time is after the deactivatingNotificationTimeout", func(t *testing.T) {
 			// given
-			murProvisionedTime := &metav1.Time{Time: time.Now().Add(-time.Duration(expectedDeactivationTimeoutOtherTier*24) * time.Hour)}
-			mur := murtest.NewMasterUserRecord(t, username, murtest.TierName(otherTier.Name), murtest.Account("cluster1"), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
+			userSignupFoobar := userSignupWithEmail(username, "foo@bar.com")
+
+			// Set usersignup state as already set to deactivating
+			states.SetDeactivating(userSignupFoobar, true)
+
+			// Set the notification status condition as sent
+			userSignupFoobar.Status.Conditions = []toolchainv1alpha1.Condition{
+				{
+					Type:               toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+					Status:             corev1.ConditionTrue,
+					LastTransitionTime: metav1.Time{Time: time.Now()},
+					Reason:             toolchainv1alpha1.UserSignupDeactivatingNotificationCRCreatedReason,
+				},
+			}
+
```

Comment:
it's just a detail, but we don't need to set the condition, right? The same applies for the following test-case

Reply:
Yeah, that's right. They're not needed. I had them set because it would be set in the actual scenario but I can remove them to make the test more concise.

Reply:
Removed in 82cc799afa2d193006a3f8611d26992908d90131